src/interface.d.ts

```diff
@@ -1,4 +1,5 @@
-export * from './components';
+// Note: this is a circular dependency which can cause issues. Is this OK?
+export * from './components'; // eslint-disable-line import/no-cycle
```

Comment:
@sslotsky what were you encountering that led to this line? This raised an issue with TS. When files import each other (circular deps), it can break webpack.

Reply:
Notice below that the library is exposing the `AuthToken` types for consumers. This is how you solve that problem in Stencil: you declare an `interface.d.ts` that exports the types from `components`, and then exports additional types.

Reply:
I actually took this from the ionic project

https://github.com/ionic-team/ionic/blob/7ae9303a979b4a1fe035f28c232320a5a920f866/core/src/interface.d.ts

Reply:
Maybe the additional types should be defined in another location and exported from here, so that component files in shadowcat can import from the other location?

Reply:
Oh I see. In that case I’ll include that link there.

I didn’t notice any issues with it fortunately—I think because it’s a TypeScript thing. And it’s been shipping in UI just fine, so probably fine to leave. But I only noticed it because ESLint pointed it out.

Reply:
I actually spent hours finding out about circular deps in our Dashboard, where 2 components imported each other and everything broke 😅

Reply:
Updated this. How does it look?